test/test.go

```diff
@@ -414,6 +417,47 @@ func (h *Helper) TempFile(path, contents string) {
 	h.Must(ioutil.WriteFile(filepath.Join(h.tempdir, path), bytes, 0644))
 }
 
+// GetTestFileBytes reads a file form the testdata directory into memory.  src is
+// relative to ./testdata.  Assumes tests take place starting in the cmd/dep
+// directory.
+func (h *Helper) GetTestFileBytes(src string) []byte {
+	content, err := ioutil.ReadFile(filepath.Join(h.origWd, "_testdata", src))
+	if err != nil {
+		panic(err)
+	}
+	if strings.HasSuffix(src, ".go") {
+		formatted, err := format.Source(content)
+		if err == nil {
+			content = formatted
+		}
+	}
+	return content
+}
+
+// GetTestFileString reads a file form the testdata directory into memory.  src is
+// relative to ./testdata.  Assumes tests take place starting in the cmd/dep
+// directory.
+func (h *Helper) GetTestFileString(src string) string {
```

Comment:
I also find the `[]byte` vs `string` thing a bit confounding. In general, I try to just follow the rule of "don't convert unless you must." The conversion isn't terribly costly, but it also isn't free, so all other things being equal, I err on the cheaper side.

(However, if you know you have to add data, `[]byte` is definitely better; strings are immutable, so e.g. string concat results in a full copy. Appending to a `[]byte` is just a normal `append`, though)